analyses/atlas-open-data-hzz/HZZ_analysis_pipeline.ipynb

```diff
@@ -278,24 +519,33 @@
   },
   {
    "cell_type": "code",
-   "execution_count": 6,
-   "id": "f451f832",
+   "execution_count": 9,
+   "id": "5435684a-1b9a-414c-b942-2f9460c8ce40",
    "metadata": {},
    "outputs": [],
    "source": [
-    "class HZZAnalysis(servicex.Analysis):\n",
+    "class HZZAnalysis(processor.ProcessorABC):\n",
     "    \"\"\"The coffea processor used in this analysis.\"\"\"\n",
     "\n",
+    "    def __init__(self):\n",
+    "        pass\n",
+    "\n",
     "    def process(self, events):\n",
     "        # type of dataset being processed, provided via metadata (comes originally from fileset)\n",
-    "        dataset_category = events.metadata[\"dataset_category\"]\n",
-    "\n",
-    "        # get lepton information from events\n",
-    "        leptons = events.lep\n",
+    "        dataset_category = events.metadata[\"dataset\"]\n",
     "\n",
     "        # apply a cut to events, based on lepton charge and lepton type\n",
-    "        events = events[lepton_filter(leptons.charge, leptons.typeid)]\n",
-    "        leptons = events.lep\n",
+    "        events = events[lepton_filter(events.lep_charge, events.lep_typeid)]\n",
+    "\n",
+    "        # construct lepton four-vectors\n",
+    "        # TODO: might be able to replace by schema\n",
+    "        leptons = ak.zip(\n",
+    "            {\"pt\": events.lep_pt,\n",
+    "             \"eta\": events.lep_eta,\n",
+    "             \"phi\": events.lep_phi,\n",
+    "             \"energy\": events.lep_energy},\n",
+    "            with_name=\"Momentum4D\",\n",
+    "        )\n",
```

Comment:
I'm not sure how this ran without a `vector.register_awkward()` in the past? I'll add it to the processor but without that I cannot get this to run.